sdk/test-utils/recorder-new/src/recorderAADCredential.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AccessToken, TokenCredential } from "@azure/core-auth";
+
+/**
+ * `TokenCredential` implementation for playback.
+ * If your regular AAD credentials don't take the recorder httpClient option, the AAD traffic won't be recorded.
+ * In this case, you'll need to bypass the AAD requests with no-op.
+ *
+ * Using this NoOpCredential as your credential in playback mode would help you bypass the AAD traffic.
+ */
+export class NoOpCredential implements TokenCredential {
```

Comment:
How about adding "Token" to make it clear that it is for token, not other types of credentials? 
```suggestion
export class NoOpTokenCredential implements TokenCredential {
```

Reply:
In practice all of our other `TokenCredential` implementers are just called "Credential" e.g. "DefaultAzureCredential", "UsernamePasswordCredential," etc.